Jetpack Connect: Implement hide free plan test

tyxla opened the pull request:
### Purpose & functionality

As @richardmuscat suggested in p7kMJG-3fL-p2, this PR implements a test where in our Jetpack signup flow we hide the Free plan, and display a Skip button at the bottom instead. 

The Skip button serves the purpose of selecting a free plan, which can do a couple of different things depending on the situation:

* If the JPC flow started in Calypso, user is redirected to `/posts/`
* If the JPC flow started in wp-admin, user is redirected to the wp-admin.

### Preview

**Before**
![](https://cldup.com/PTPx2DMkh8.png)

**After**
![](https://cldup.com/SKR0PTnRfd.png)

### To test

* Checkout this branch or get it going on calypso.live
* Enable the test by typing the following in your console: `localStorage.setItem( 'ABTests', '{"jetpackConnectHideFreePlan_20170905":"hide"}' );`
* Connect a new Jetpack site and get to the Plan selection page.
* Verify you no longer see the free plan.
* Verify you see the Skip link at the bottom.
* Click the Skip link at the bottom and verify it works just like the Free plan button does.
* Disable the test by typing the following in your console: `localStorage.setItem( 'ABTests', '{"jetpackConnectHideFreePlan_20170905":"show"}' );`
* Test various flows with this AB test enabled and verify it works as expected.
* Connect a new Jetpack site and get to the Plan selection page.
* Verify you see the Free plan and you don't see the Skip link.
* Test various flows with this AB test disabled and verify it works as expected.

### Questions

* Do we need to apply any additional styling to the Skip link? cc @rickybanister 
* Do we want to update the Skip button copy in any way? cc @Automattic/editorial

michelleweber commented:
There are lots of things you could try -- Skip, Skip This Step, Choose Later, Start Free. Skip's fine to start with!

richardmuscat commented:
@tyxla -- fantastic stuff. it works for a new site, but...

i'd like to to work whenever anyone visits the plan screen even if its not a new site

atm, if i test it on https://calypso.live/jetpack/connect/store?branch=try/jetpack-connect-hide-free-plan-test it doesn't seem to work

richardmuscat commented:
re "skip", it looks good to me but I'd make the text a bit larger: same size as the text on the "Start with [plan]" buttons

johnHackworth commented:
codewise, this looks good :+1:

fditrapani commented on client/jetpack-connect/plans.jsx:
```diff
@@ -291,7 +302,23 @@ class Plans extends Component {
+				>
+					{
+						hideFreePlanTest && (
+							<div className="jetpack-connect__plans-nav-buttons">
```
This is the component we use for signup: https://github.com/Automattic/wp-calypso/blob/master/client/signup/step-wrapper/index.jsx#L39.

Not sure if it's compatible or worth replacing?

tyxla replied:
I was also thinking about using that component, but it was adding too much margin, and now with the font size alterations it's creating more trouble than helping.

fditrapani replied:
No problem. Sounds good.

tyxla commented:
> Skip's fine to start with!

Thanks @michelleweber!

> i'd like to to work whenever anyone visits the plan screen even if its not a new site

@richardmuscat, I wasn't sure if you'll actually want to include these flows in the test. I've now updated the PR to apply for all of the landing page cases, too.

> re "skip", it looks good to me but I'd make the text a bit larger: same size as the text on the "Start with [plan]" buttons

Makes sense to me. Updated the PR to reflect that.

This one could use another look, @richardmuscat @johnHackworth @fditrapani, thanks!

fditrapani commented:
Updates look good 👍

rickybanister commented:
Looks good to me as well. My only feedback was going to be the same as @richardmuscat—that I would like the see the new plan icons and the rest of the display on the post-signup plans page as well.

tyxla commented:
> I would like the see the new plan icons and the rest of the display on the post-signup plans page as well.

That can be a different test that we can run after the GM, sounds good @rickybanister?

richardmuscat commented:
@tyxla -- lgtm. 🚢 it

tyxla commented:
Thank you folks!
